consensus/src/config.rs

```diff
@@ -66,6 +66,11 @@ pub fn is_emergency_iter(iter: u8) -> bool {
     iter >= EMERGENCY_MODE_ITERATION_THRESHOLD
 }
 
+/// Returns if the next iteration generator needs to be excluded
+pub fn exclude_next_generator(iter: u8) -> bool {
```

Comment:
Maybe `last_step_running`  should reuse this to avoid duplication?

Reply:
no, it's not last step, this applies to validation step too

Reply:
I mean, you can reuse "exclude_next_generator/is_last_iteration" in `ExecutionCtx::last_step_running` that already exists.

Reply:
```rust
    /// Returns true if the last step of last iteration is currently running
    fn last_step_running(&self) -> bool {
        self.iteration == CONSENSUS_MAX_ITER - 1
            && self.step_name() == StepName::Ratification
    }
    ```
    
There is no `is_last_iteration`

Btw, if you mean to extract `self.iteration == CONSENSUS_MAX_ITER - 1` into a fn called `is_last_iteration` I agree.
I'll tackle it right after this PR